Source/ACE/Network/GameMessages/Messages/GameMessageFellowshipFullUpdate.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using ACE.Common.Extensions;
+using ACE.Entity.Actions;
+using ACE.Entity;
+
+namespace ACE.Network.GameMessages.Messages
+{
+    public class GameMessageFellowshipFullUpdate : GameMessage
+    {
+        public GameMessageFellowshipFullUpdate(Session session)
+            : base(GameMessageOpcode.GameEvent, GameMessageGroup.Group09)
+        {
+            Fellowship fellowship = session.Player.Fellowship;
+            ActionChain chain = new ActionChain();
+            chain.AddAction(session.Player, () =>
+            {
+                Writer.Write(session.Player.Guid.Full);
+                Writer.Write(session.GameEventSequence++);
+                Writer.Write((uint)GameEvent.GameEventType.FellowshipFullUpdate);
+
+                // the current number of fellowship members
+                Writer.Write((UInt16)fellowship.FellowshipMembers.Count);
+
+                // ????
+                Writer.Write((byte)0x10);
+                Writer.Write((byte)0x00);
+
+                // --- FellowInfo ---
+                
+                foreach (Player fellow in fellowship.FellowshipMembers)
+                {
+                    if (fellow.Guid.Full != session.Player.Guid.Full)
+                    {
+                        chain.AddAction(fellow, () => { WriteFellow(fellow); });
```

Comment:
I just realized, I misunderstood your comment last night, and probably didn't provide the information you needed.

ActionChains are designed to sequence "Actions" which should run one after the other.  In this instance, your adding to the chain while its running.  This is messy business and probably doesn't do what you want.

Instead consider something like this:

```cs
chain.AddAction(session.Player, () =>
{
    ... Write up until the fellowship info foreach ...
    ActionChain internalChain = new ActionChain();
    foreach (Player fellow in fellowhsip.FellowshipMembers)
    {
        internalChain.AddAction(fellow, () => WriteFellow(fellow) );
    }
    internalChain.AddAction(session.Player, () =>
    {
        ... Write fellow name through end ...
    }
    internalChain.EnqueueChain();
}
chain.EnqueueChain();
```

This will cause internalChain to be created when the action in chain is actually run.

There's a larger question here of when the "fellowship" work should be done (the fellowship itself is a structure which can be concurrently accessed by other players -- at what point should its work be scheduled so we guarantee it doesn't race with other accesses/modifications to the fellowhship).